crates/proof-of-sql/src/base/database/column_index_operation.rs

```diff
@@ -0,0 +1,150 @@
+use super::{slice_operation::apply_slice_to_indexes, Column, ColumnOperationResult, ColumnType};
+use crate::base::scalar::Scalar;
+use bumpalo::Bump;
+
+/// Apply a `Column` to a vector of indexes, returning a new `Column` with the
+/// values at the given indexes. Repetitions are allowed.
+///
+/// # Panics
+/// Panics if any of the indexes are out of bounds.
+#[allow(dead_code)]
+pub(crate) fn apply_column_to_indexes<'a, S>(
+    column: &Column<'a, S>,
+    alloc: &'a Bump,
+    indexes: &[usize],
+) -> ColumnOperationResult<Column<'a, S>>
```

Comment:
NIT: This comment might not be useful depending on how this function will be used later. Without that context, I'd say it would be better if the error type was just a new `IndexOutOfBounds` error type since most of the column operation errors cannot occur in this function. But if this function is only ever used in contexts that are dealing with that error type, then it's probably fine.